src/python/pants/backend/jvm/subsystems/scoverage_platform.py

```diff
@@ -0,0 +1,125 @@
+# Copyright 2015 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import os
+from pants.build_graph.injectables_mixin import InjectablesMixin
+from pants.subsystem.subsystem import Subsystem
+from pants.java.jar.jar_dependency import JarDependency
+from pants.backend.jvm.targets.jar_library import JarLibrary
+from pants.build_graph.address import Address
+from typing import Any, Dict, List, Optional, Union
+
+
+SCOVERAGE = "scoverage"
+blacklist_file = 'new_blacklist_scoverage'
+
+
+class ScoveragePlatform(InjectablesMixin, Subsystem):
+  """The scala coverage platform."""
+
+  options_scope = 'scoverage'
+
+  @classmethod
+  def register_options(cls, register):
+    super().register_options(register)
+    register('--enable-scoverage',
+      default=False,
+      type=bool,
+      help='Specifies whether to generate scoverage reports for scala test targets.'
```

Comment:
If we must have two options to enable scoverage, then I think that they should refer to one another: ie, this option should indicate that the `--test-junit-coverage-processor=scoverage` value must be set as well, and vice versa.

Reply:
So I have currently set it to raise an error if `--test-junit-coverage-processor=scoverage` is specified but scoverage is not enabled. Also, is `--scoverage-enable-scoverage` is enabled, the former gets set to scoverage processor in `manager.py` . Would that work?